File: Python_Codes/7. Exception Handling/Basic.py
class ArithmeticExpression:
    def arithmetic_exception(self):
        try:
            # This will cause a ZeroDivisionError if b is 0
            a = int(input("Enter the numerator: "))
            b = int(input("Enter the denominator: "))
            result = a / b
            print("Result:", result)
        except ZeroDivisionError as e:
            # Exception caught and handled here
            print("Caught a ZeroDivisionError:", e)
        finally:
            # This block always executes
            print("Finally block executed.")


def main():
    a1 = ArithmeticExpression()
    a1.arithmetic_exception()


if __name__ == "__main__":
    main()

File: Projects/JavaProject/src/com/Exception_Handling/Basic.java
package com.Exception_Handling;

import java.util.Scanner;

class ArithmeticExpression {
    public void arithmeticException() {
        try {
            // This will cause an ArithmeticException (division by zero)
            Scanner sc = new Scanner(System.in);
            int a = sc.nextInt();
            int b = sc.nextInt();
            int result = a / b;
            System.out.println("Result: " + result);
            sc.close();
        } 
        
        catch (ArithmeticException e) {
            // Exception caught and handled here
            System.out.println("Caught an ArithmeticException: " + e.getMessage());
        } 
        
        finally {
            // This block always executes, whether an exception is thrown or not
            System.out.println("Finally block executed.");
        }
    }
}

public class Basic {
    public static void main(String[] args) {
        ArithmeticExpression a1 = new ArithmeticExpression();
        a1.arithmeticException();
    }
}

